### PR TITLE
Migrate Vault-backend plugin to new backend system

### PR DESCRIPTION
**Pull request by SaachiNayyer**
#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/vault-backend/README.md [hunk 1]
```diff
@@ -8,6 +8,20 @@ Vault is an identity-based secrets and encryption management system. A secret is
...
+
+In your `packages/backend/src/index.ts` make the following changes:
+
+```diff
+  import { createBackend } from '@backstage/backend-defaults';
+  const backend = createBackend();
+  // ... other feature additions
+  backend.add(import('@backstage/plugin-vault-backend'));
+  backend.start();
+```
```
Comment (vinzscam) [inline plugins/vault-backend/README.md:23]:
```suggestion
```diff
  import { createBackend } from '@backstage/backend-defaults';
  const backend = createBackend();
  // ... other feature additions
+ backend.add(import('@backstage/plugin-vault-backend'));
  backend.start();
```
```

**Comment by ivangonzalezacuna** [comment]
Oh, turns out I was also working on this and, when I started, this PR wasn't published yet :sweat_smile: 

I think this PR https://github.com/backstage/backstage/pull/20315 should contain more customizations available for the plugin.

But if it's better to start with this one, I could clone mine on the meantime

File: plugins/vault-backend/README.md (hunk 1 above)
Comment (vinzscam) [reply plugins/vault-backend/README.md:23]: sorry yeah it wasn't so clear 🙈 I've just added a `+` on line 21 to mark the changes that adopters need to be done 😅

Comment (SaachiNayyer) [reply plugins/vault-backend/README.md:23]: Hi [vinzscam](https://github.com/vinzscam)
 I have tried to address the comment, if there are any other please let me know.

Comment (SaachiNayyer) [reply plugins/vault-backend/README.md:23]: Hi [vinzscam](https://github.com/vinzscam)
Are there any updates for this PR

Thanks

**Comment by SaachiNayyer** [comment]
Hi [vinzscam](https://github.com/vinzscam), [freben](https://github.com/freben) & [ivangonzalezacuna](https://github.com/ivangonzalezacuna),

 

I'm happy to close this PR if it means that @ivangonzalezacuna can get his PR #20315 merged as it contains additional extra improvements to this PR. 

Can a maintainer of backstage let me know if doing this would make it easier to get that PR merged then I will do it?

 

Many Thanks,

Saachi

**Comment by freben** [comment]
Closing as superseded by https://github.com/backstage/backstage/pull/20315. Thank you so much though! And if this was in error, let us know

**Pull request closed without merging**